ReactWindows/ReactNative/Views/Web/ReactWebViewManager.cs

```diff
@@ -0,0 +1,306 @@
+        public override void ReceiveCommand(WebView view, int commandId, JArray args)
+        {
+            switch (commandId)
+            {
+                case CommandGoBack:
+                    if (view.CanGoBack) view.GoBack();
+                    break;
+                case CommandGoForward:
+                    if (view.CanGoForward) view.GoForward();
+                    break;
+                case CommandReload:
+                    view.Refresh();
+                    break;
+                default:
+                    throw new InvalidOperationException(
+                        $"Unsupported command '{commandId}' received by '{typeof(ReactWebViewManager)}'.");
+            }
+        }
+
+        /// <summary>
+        /// Called when view is detached from view hierarchy and allows for 
+        /// additional cleanup by the <see cref="ReactWebViewManager"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, WebView view)
+        {
+            view.NavigationCompleted -= OnNavigationCompleted;
+            view.NavigationStarting -= OnNavigationStarting;
+        }
+
+        /// <summary>
+        /// Creates a new view instance of type <see cref="WebView"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <returns>The view instance.</returns>
+        protected override WebView CreateViewInstance(ThemedReactContext reactContext)
+        {
+            return new WebView();
+        }
+
+        /// <summary>
+        /// Subclasses can override this method to install custom event 
+        /// emitters on the given view.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view instance.</param>
+        protected override void AddEventEmitters(ThemedReactContext reactContext, WebView view)
+        {
+            view.NavigationCompleted += OnNavigationCompleted;
+            view.NavigationStarting += OnNavigationStarting;
+        }
+
+        private void OnNavigationCompleted(object sender, WebViewNavigationCompletedEventArgs e)
+        {
+            var webView = (WebView)sender;
+            var reactContext = webView.GetReactContext();
+
+            if (e.IsSuccess)
+            {
+                var script = default(string);
+
+                if (_injectedJS.TryGetValue(webView.GetTag(), out script) && script.Length > 0)
+                {
+                    string[] args = { script };
+                    RunOnDispatcher(async () =>
+                    {
+                        try
+                        {
+                            await webView.InvokeScriptAsync("eval", args);
+                        }
+                        catch (Exception)
```

Comment:
Whatever android and ios does.

---
In reply to: [60720727](https://github.com/ReactWindows/react-native/pull/372#discussion_r60720727) [](ancestors = 60720727)